contributors/design-proposals/deploy.md

```diff
@@ -118,29 +118,29 @@ See issue [#17164](https://github.com/kubernetes/kubernetes/issues/17164).
+ReplicaSets. A Deployment obtains the largest version number from its ReplicaSets and increments the number by 1 upon update or creation of the Deployment, and updates the version annotation of its new ReplicaSet.
 
-When the users perform a rollback, i.e. `kubectl rollout undo`, the deployment first looks at its existing replication controllers, regardless of their number of replicas. Then it finds the one with annotation `rollout.kubectl.kubernetes.io/version` that either contains the specified rollback version number or contains the second largest version number among all the replication controllers (current new replication controller should obtain the largest version number) if the user didn't specify any version number (the user wants to rollback to the last change). Lastly, it
-starts scaling up that replication controller it's rolling back to, and scaling down the current ones, and then update the version counter and the rollout annotations accordingly.
+When the users perform a rollback, i.e. `kubectl rollout undo`, the Deployment first looks at its existing ReplicaSets, regardless of their number of replicas. Then it finds the one with annotation `deployment.kubernetes.io/revision` that either contains the specified rollback version number or contains the second largest version number among all the ReplicaSets (current new ReplicaSet should obtain the largest version number) if the user didn't specify any version number (the user wants to rollback to the last change). Lastly, it
+starts scaling up that ReplicaSet it's rolling back to, and scaling down the current ones, and then update the version counter and the rollout annotations accordingly.
 
-Note that a deployment's replication controllers use PodTemplate hashes (i.e. the hash of `.spec.template`) to distinguish from each others. When doing rollout or rollback, a deployment reuses existing replication controller if it has the same PodTemplate, and its `rollout.kubectl.kubernetes.io/change-source` and `rollout.kubectl.kubernetes.io/version` annotations will be updated by the new rollout. At this point, the earlier state of this replication controller is lost in history. For example, if we had 3 replication controllers in
-deployment history, and then we do a rollout with the same PodTemplate as version 1, then version 1 is lost and becomes version 4 after the rollout.
+Note that a Deployment's ReplicaSets use PodTemplate hashes (i.e. the hash of `.spec.template`) to distinguish from each others. When doing rollout or rollback, a Deployment reuses existing ReplicaSet if it has the same PodTemplate, and its `kubernetes.io/change-cause` and `deployment.kubernetes.io/revision` annotations will be updated by the new rollout. At this point, the earlier state of this ReplicaSet is lost in history. For example, if we had 3 ReplicaSets in
+Deployment history, and then we do a rollout with the same PodTemplate as version 1, then version 1 is lost and becomes version 4 after the rollout.
```

Comment:
We have added an annotation that holds the history of the revisions a ReplicaSet has served. Mind referencing it here?
https://github.com/kubernetes/kubernetes/pull/34249